clients/src/main/java/org/apache/kafka/common/security/scram/ScramMechanism.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.security.scram;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public enum ScramMechanism {
+
+    SCRAM_SHA_224("SCRAM-SHA-224", "HmacSHA224"),
+    SCRAM_SHA_256("SCRAM-SHA-256", "HmacSHA256"),
+    SCRAM_SHA_384("SCRAM-SHA-384", "HmacSHA384"),
+    SCRAM_SHA_512("SCRAM-SHA-512", "HmacSHA512");
+
+    private final String mechanismName;
+    private final String hashAlgorithm;
+    private final String macAlgorithm;
+
+    private static final String[] ALL_MECHANISMS;
+    private static final Map<String, ScramMechanism> MECHANISMS_MAP;
+
+    static {
+        ScramMechanism[] values = values();
+        ALL_MECHANISMS = new String[values.length];
+        Map<String, ScramMechanism> map = new HashMap<>();
+        for (int i = 0; i < ALL_MECHANISMS.length; i++) {
+            ScramMechanism mech = values[i];
+            ALL_MECHANISMS[i] = mech.mechanismName;
+            map.put(mech.mechanismName, mech);
+        }
+        MECHANISMS_MAP = Collections.unmodifiableMap(map);
+    }
+
+    ScramMechanism(String mechanismName, String macAlgorithm) {
+        this.mechanismName = mechanismName;
+        String scramPrefix = "SCRAM-";
+        if (!mechanismName.startsWith(scramPrefix))
+            throw new IllegalArgumentException("Invalid mechanism " + mechanismName);
+        this.hashAlgorithm = mechanismName.substring(scramPrefix.length());
```

Comment:
Nit: instead of passing the full string and then doing a substring on it, I think the other way around is a little bit clearer. That is, pass the components of the mechanism name separately and join them in the constructor.